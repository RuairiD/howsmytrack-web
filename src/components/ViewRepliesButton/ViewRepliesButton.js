import React from 'react';

import { Badge, Button, Icon } from 'antd';

type Props = {
    replies: number,
    unreadReplies: number,
    onClick: () => {},
};

class ViewRepliesButton extends React.Component<Props> {
    /*
     * Component for rendering a button for opening a FeedbackResponseRepliesModal
     * Indicates how many replies there are to read and how many are unread.
     */
    renderRepliesText = () => {
        if (this.props.replies === 0) {
            return "Leave a Reply";
        }

        let repliesText = "View " + this.props.replies + " replies";
        if (this.props.replies === 1) {
            repliesText = "View 1 reply";
        }
        
        return repliesText;
    };

    render() {
        return (
            <div style={{ paddingRight: '20px' }}>
                <Badge count={this.props.unreadReplies}>
                    <Button onClick={this.props.onClick}>
                        <Icon type="message" />
                        {this.renderRepliesText()}
                    </Button>
                </Badge>
            </div>
        );
    }
}

export default ViewRepliesButton;
