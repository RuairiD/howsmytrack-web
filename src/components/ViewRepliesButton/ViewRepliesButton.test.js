import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import ViewRepliesButton from "./ViewRepliesButton";

describe("ViewRepliesButton", () => {
    it("displays a prompt to write a reply if none have been posted already", () => {
        const wrapper = shallow(
            <ViewRepliesButton
                replies={0}
                unreadReplies={0}
                onClick={() => {}}
            />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("displays a singular string for 1 reply", () => {
        const wrapper = shallow(
            <ViewRepliesButton
                replies={1}
                unreadReplies={0}
                onClick={() => {}}
            />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("displays a plural string for multiple replies", () => {
        const wrapper = shallow(
            <ViewRepliesButton
                replies={2}
                unreadReplies={0}
                onClick={() => {}}
            />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("displays a badge with a singular title/hovertext indicating one unread reply", () => {
        const wrapper = shallow(
            <ViewRepliesButton
                replies={2}
                unreadReplies={1}
                onClick={() => {}}
            />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("displays a badge with a plural title/hovertext indicating multiple unread replies", () => {
        const wrapper = shallow(
            <ViewRepliesButton
                replies={2}
                unreadReplies={2}
                onClick={() => {}}
            />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
