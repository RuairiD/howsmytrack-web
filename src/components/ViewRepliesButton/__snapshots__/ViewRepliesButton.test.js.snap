// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewRepliesButton displays a badge with a plural title/hovertext indicating multiple unread replies 1`] = `
<Button
  block={true}
  ghost={false}
  htmlType="button"
  loading={false}
  onClick={[Function]}
  style={
    Object {
      "marginTop": "0.5em",
    }
  }
>
  <Icon
    type="message"
  />
  View 2 replies
  <Badge
    count={2}
    dot={false}
    overflowCount={99}
    showZero={false}
    style={
      Object {
        "marginLeft": "0.5em",
      }
    }
    title="2 unread replies"
  />
</Button>
`;

exports[`ViewRepliesButton displays a badge with a singular title/hovertext indicating one unread reply 1`] = `
<Button
  block={true}
  ghost={false}
  htmlType="button"
  loading={false}
  onClick={[Function]}
  style={
    Object {
      "marginTop": "0.5em",
    }
  }
>
  <Icon
    type="message"
  />
  View 2 replies
  <Badge
    count={1}
    dot={false}
    overflowCount={99}
    showZero={false}
    style={
      Object {
        "marginLeft": "0.5em",
      }
    }
    title="1 unread reply"
  />
</Button>
`;

exports[`ViewRepliesButton displays a plural string for multiple replies 1`] = `
<Button
  block={true}
  ghost={false}
  htmlType="button"
  loading={false}
  onClick={[Function]}
  style={
    Object {
      "marginTop": "0.5em",
    }
  }
>
  <Icon
    type="message"
  />
  View 2 replies
  <Badge
    count={0}
    dot={false}
    overflowCount={99}
    showZero={false}
    style={
      Object {
        "marginLeft": "0.5em",
      }
    }
    title="0 unread replies"
  />
</Button>
`;

exports[`ViewRepliesButton displays a prompt to write a reply if none have been posted already 1`] = `
<Button
  block={true}
  ghost={false}
  htmlType="button"
  loading={false}
  onClick={[Function]}
  style={
    Object {
      "marginTop": "0.5em",
    }
  }
>
  <Icon
    type="message"
  />
  Leave a Reply
  <Badge
    count={0}
    dot={false}
    overflowCount={99}
    showZero={false}
    style={
      Object {
        "marginLeft": "0.5em",
      }
    }
    title="0 unread replies"
  />
</Button>
`;

exports[`ViewRepliesButton displays a singular string for 1 reply 1`] = `
<Button
  block={true}
  ghost={false}
  htmlType="button"
  loading={false}
  onClick={[Function]}
  style={
    Object {
      "marginTop": "0.5em",
    }
  }
>
  <Icon
    type="message"
  />
  View 1 reply
  <Badge
    count={0}
    dot={false}
    overflowCount={99}
    showZero={false}
    style={
      Object {
        "marginLeft": "0.5em",
      }
    }
    title="0 unread replies"
  />
</Button>
`;
